File: demos/jans-chip/android/app/src/main/java/io/jans/chip/dao/OPConfigurationDao.java
package io.jans.chip.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import io.jans.chip.modal.OPConfiguration;
@Dao
public interface OPConfigurationDao {
    @Insert
    void insert(OPConfiguration opConfiguration);

    @Update
    void update(OPConfiguration opConfiguration);

    @Query("SELECT * FROM OP_CONFIGURATION")
    List<OPConfiguration> getAll();

    @Query("DELETE FROM OP_CONFIGURATION")
    void deleteAll();
}

File: demos/jans-chip/android/app/src/main/java/io/jans/chip/dao/OPConfigurationDao.kt
package io.jans.chip.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import io.jans.chip.model.OPConfiguration

@Dao
interface OPConfigurationDao {
    @Insert
    fun insert(opConfiguration: OPConfiguration?)

    @Update
    fun update(opConfiguration: OPConfiguration?)

    @Query("SELECT * FROM OP_CONFIGURATION")
    fun getAll(): List<OPConfiguration>

    @Query("DELETE FROM OP_CONFIGURATION")
    fun deleteAll()
}
